test/e2e/density.go

```diff
@@ -352,27 +356,38 @@ var _ = framework.KubeDescribe("Density", func() {
 		podsPerNode    int
 		// Controls how often the apiserver is polled for pods
 		interval time.Duration
+		// What kind of resource we should be creating. Default: ReplicationController
+		kind schema.GroupKind
 	}
 
 	densityTests := []Density{
 		// TODO: Expose runLatencyTest as ginkgo flag.
-		{podsPerNode: 3, runLatencyTest: false},
-		{podsPerNode: 30, runLatencyTest: true},
-		{podsPerNode: 50, runLatencyTest: false},
-		{podsPerNode: 95, runLatencyTest: true},
-		{podsPerNode: 100, runLatencyTest: false},
+		{podsPerNode: 3, runLatencyTest: false, kind: api.Kind("ReplicationController")},
+		{podsPerNode: 30, runLatencyTest: true, kind: api.Kind("ReplicationController")},
+		{podsPerNode: 50, runLatencyTest: false, kind: api.Kind("ReplicationController")},
+		{podsPerNode: 95, runLatencyTest: true, kind: api.Kind("ReplicationController")},
+		{podsPerNode: 100, runLatencyTest: false, kind: api.Kind("ReplicationController")},
 	}
 
 	for _, testArg := range densityTests {
 		feature := "ManualPerformance"
+		if testArg.kind.Empty() {
+			testArg.kind = api.Kind("ReplicationController")
+		}
+		if testArg.kind != extensions.Kind("Deployment") && testArg.kind != extensions.Kind("ReplicaSet") && testArg.kind != api.Kind("ReplicationController") {
```

Comment:
Let's remove this if - it shouldn't be needed to check if test is written correctly :)